package mammut.energy;
option optimize_for = LITE_RUNTIME;

message CountersCpuGet{
}

message CountersCpuGetRes{
    message CounterCpu{
        required uint32 cpu_id = 1;
        required bool has_graphic = 2;
        required bool has_dram = 3;
    }
    repeated CounterCpu counters = 1;
}

message CounterCpuReset{
    required uint32 cpu_id = 1;
}

message CounterCpuResetRes{
}

enum CounterCpuType {
	COUNTER_CPU_TYPE_ALL = 0;
    COUNTER_CPU_TYPE_CPU = 1;
    COUNTER_CPU_TYPE_CORES = 2;
    COUNTER_CPU_TYPE_GRAPHIC = 3;
    COUNTER_CPU_TYPE_DRAM = 4;
}

message CounterCpuGetJoules{
    required uint32 cpu_id = 1;
    required CounterCpuType type = 2;
}

message CounterCpuGetJoulesRes{
    required double joules = 1;
}

message CounterCpuGetJoulesAllRes{
    required double joules_cpu = 1;
    required double joules_cores = 2;
    required double joules_graphic = 3;
    required double joules_dram = 4;
}